lib/rust/prelude/src/data/non_empty_vec.rs

```diff
@@ -320,75 +343,64 @@ impl<T> NonEmptyVec<T> {
     ///
     /// ```
     /// use enso_prelude::NonEmptyVec;
-    /// let mut vec = NonEmptyVec::new(0, vec![1, 2, 3, 4, 5]);
+    /// let mut vec = NonEmptyVec::<_, usize>::new(0, vec![1, 2, 3, 4, 5]);
     /// let replacements = [10, 20, 30, 40];
     /// let yielded: Vec<_> = vec.splice(..2, replacements.iter().cloned()).collect();
     /// assert_eq!(vec.as_slice(), &[10, 20, 30, 40, 2, 3, 4, 5]);
     /// assert_eq!(yielded, &[0, 1])
     /// ```
-    pub fn splice<R, I>(
+    pub fn splice<R, Iter>(
         &mut self,
         range: R,
-        replace_with: I,
-    ) -> Splice<<I as IntoIterator>::IntoIter>
+        replace_with: Iter,
+    ) -> Splice<<Iter as IntoIterator>::IntoIter>
     where
-        I: IntoIterator<Item = T>,
-        R: RangeBounds<usize>,
+        Iter: IntoIterator<Item = T>,
+        R: RangeBounds<I>,
     {
         self.elems.splice(range, replace_with)
     }
```

Comment:
Not a regression, but: This API breaks the contract of `NonEmptyVec`. You can do: `vec.splice(.., None);` to get an empty `NonEmptyVec`.

Reply:
Wait, this will return `Splice`, not `NonEmptyVec`. I feel Im missing something here.

Reply:
Oh, good! I was missing something.

Reply:
@wdanilo Actually, the problem is not the return value, it's the side effect on `self`. This test case would fail:

```rust
#[test]
fn empty_nonempty_via_splice() {
    let mut vec = NonEmptyVec::<_>::new(0, vec![]);
    vec.splice(.., None);
    assert_ne!(NonEmptyVec::len(&vec), 0);
}
```

Reply:
Oh, I was totally blind not to see it. I removed this function from NonEmptyVector. It seems no one used it.